apps/openmw/mwclass/mobile.cpp

```diff
@@ -0,0 +1,87 @@
+#include "mobile.hpp"
+
+#include <components/esm/loadmgef.hpp>
+
+#include "../mwbase/environment.hpp"
+#include "../mwbase/world.hpp"
+#include "../mwbase/mechanicsmanager.hpp"
+#include "../mwbase/soundmanager.hpp"
+
+#include "../mwmechanics/creaturestats.hpp"
+#include "../mwmechanics/movement.hpp"
+#include "../mwmechanics/magiceffects.hpp"
+
+#include "../mwphysics/physicssystem.hpp"
+
+#include "../mwworld/inventorystore.hpp"
+
+namespace MWClass
+{
+    Mobile::Mobile() {}
+
+    Mobile::~Mobile() {}
+
+    void Mobile::adjustPosition(const MWWorld::Ptr& ptr, bool force) const
+    {
+        MWBase::Environment::get().getWorld()->adjustPosition(ptr, force);
+    }
+
+    void Mobile::insertObject(const MWWorld::Ptr& ptr, const std::string& model, MWPhysics::PhysicsSystem& physics) const
+    {
+        if (!model.empty())
+        {
+            physics.addActor(ptr, model);
+            if (getCreatureStats(ptr).isDead())
+                MWBase::Environment::get().getWorld()->enableActorCollision(ptr, false);
+        }
+        MWBase::Environment::get().getMechanicsManager()->add(ptr);
+    }
+
+    void Mobile::block(const MWWorld::Ptr &ptr) const
+    {
+        MWWorld::InventoryStore& inv = getInventoryStore(ptr);
+        MWWorld::ContainerStoreIterator shield = inv.getSlot(MWWorld::InventoryStore::Slot_CarriedLeft);
+        if (shield == inv.end())
+            return;
+
+        MWBase::SoundManager *sndMgr = MWBase::Environment::get().getSoundManager();
+        switch (shield->getClass().getEquipmentSkill(*shield))
+        {
+        case ESM::Skill::LightArmor:
+            sndMgr->playSound3D(ptr, "Light Armor Hit", 1.0f, 1.0f);
+            break;
+        case ESM::Skill::MediumArmor:
+            sndMgr->playSound3D(ptr, "Medium Armor Hit", 1.0f, 1.0f);
+            break;
+        case ESM::Skill::HeavyArmor:
+            sndMgr->playSound3D(ptr, "Heavy Armor Hit", 1.0f, 1.0f);
+            break;
+        default:
+            return;
+        }
```

Comment:
Can we keep the standard indenting style please? I know we aren't super consistent with switch-case, but that doesn't mean we need to make it worse when refactoring.